test/e2e/e2e_suite_test.go

```diff
@@ -17,104 +16,175 @@ limitations under the License.
-	RegisterFailHandler(Fail)
-	_, _ = fmt.Fprintf(GinkgoWriter, "Starting etcd-operator integration test suite\n")
-	RunSpecs(t, "e2e suite")
+func TestMain(m *testing.M) {
+	testEnv = env.New()
+	kindClusterName := "etcd-cluster"
+	kindCluster := kind.NewCluster(kindClusterName)
+
+	log.Println("Creating KinD cluster...")
+	testEnv.Setup(
+		envfuncs.CreateCluster(kindCluster, kindClusterName),
+		envfuncs.CreateNamespace(namespace),
+
+		// install cluster dependencies mgr
+		func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+			log.Println("Installing prometheus operator...")
+			if p := utils.RunCommand(fmt.Sprintf("kubectl apply -f %s --server-side", promUrl)); p.Err() != nil {
+				log.Printf("Failed to deploy prometheus: %s: %s", p.Err(), p.Out())
+				return ctx, p.Err()
+			}
+
+			log.Println("Installing cert-manager...")
+			client := cfg.Client()
+
+			if p := utils.RunCommand(fmt.Sprintf("kubectl apply -f %s", certMgrUrl)); p.Err() != nil {
+				log.Printf("Failed to deploy cert-manager: %s: %s", p.Err(), p.Out())
+				return ctx, p.Err()
+			}
+
+			// wait for CertManager to be ready
+			log.Println("Waiting for cert-manager deployment to be available...")
+			if err := wait.For(
+				conditions.New(client.Resources()).DeploymentAvailable("cert-manager-webhook", "cert-manager"),
+				wait.WithTimeout(5*time.Minute),
+				wait.WithInterval(10*time.Second),
+			); err != nil {
+				log.Printf("Timedout while waiting for cert-manager deployment: %s", err)
+				return ctx, err
+			}
+			return ctx, nil
+		},
+
+		// install tool dependencies
+		func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+			log.Println("Installing bin tools...")
+			if p := utils.RunCommand(
+				fmt.Sprintf("go install sigs.k8s.io/kustomize/kustomize/v5@%s", kustomizeVer),
+			); p.Err() != nil {
+				log.Printf("Failed to install kustomize binary: %s: %s", p.Err(), p.Out())
+				return ctx, p.Err()
+			}
+			if p := utils.RunCommand(
+				fmt.Sprintf("go install sigs.k8s.io/controller-tools/cmd/controller-gen@%s", ctrlGenVer),
+			); p.Err() != nil {
+				log.Printf("Failed to install controller-gen binary: %s: %s", p.Err(), p.Out())
+				return ctx, p.Err()
+			}
+			return ctx, nil
+		},
+
+		// generate and deploy resource configurations
+		func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+			log.Println("Building source components...")
+			origWd, _ := os.Getwd()
+
+			// change dir for Make file or it will fail
+			if err := os.Chdir("../../"); err != nil {
+				log.Printf("Unable to set working directory: %s", err)
+				return ctx, err
+			}
+
+			// gen manifest files
+			log.Println("Generate manifests...")
+			if p := utils.RunCommand(
+				`controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases`,
+			); p.Err() != nil {
+				log.Printf("Failed to generate manifests: %s: %s", p.Err(), p.Out())
+				return ctx, p.Err()
+			}
```

Comment:
The Makefile already has several targets: `install`, `manifests`, `kustomize`. Are you going cleanup all such targets in Makefile later?  It isn't clear what's your intention.